### ISSUE TITLE
Please add audio recorder!

### ISSUE THREAD

**Issue by cdavidyoung** [jfversluis/Plugin.Maui.Audio#33]
I don't see one for Maui yet.  However, I saw a comment that you were planning to add one.  I suppose I should volunteer to do it ...

**Comment by jfversluis** [comment]
Coordinate with @bijington he has already done some work on it I think. It's coming!

**Comment by cdavidyoung** [comment]
Great!  I am available for testing on Windows, Android, MacCatalyst.  The iPhone won't deploy because of some certificate issue.

**Comment by cdavidyoung** [comment]
Any progress on this? I can now test on all 4 platforms.

**Comment by cdavidyoung** [comment]
I see Shaun's blog at https://blog.bijington.com/ but I don't see how to message him.  I don't do twitter.

Is there anything I could do to help move this forward?  I have developed both a Xamarin and now Maui app but have not worked under the hood of a Nuget package yet.  I would be willing to learn.

**Comment by cdavidyoung** [comment]
Hi Shaun.  On the Xamarin side the functionality was split between
AudioRecorderService (Plugin.AudioRecorder) and AudioPlayer.  Splitting the
functionality on the Maui side is fine with me if that makes it any easier.

Regarding https://github.com/NateRickard/Plugin.AudioRecorder, what would
it take to convert that to Maui even if it only worked with Android and
iPhone?  That would be a start and it would allow me to publish my mobile
apps.

Thanks for your help!  If I get desperate I may look at doing the
conversion myself but I have yet to work on a package like that and know
what it takes to convert from Xamarin to Maui.

Charles

**Comment by cdavidyoung** [comment]
So I am a little confused.  Is there something I can test or will a nuget solution be released soon?

**Comment by jfversluis** [comment]
Nothing with this functionality has ben released yet. If you want to test it right now you can get the code from here and test that: https://github.com/jfversluis/Plugin.Maui.Audio/pull/35

It's still a work in progress.

**Comment by hercul1017** [comment]
Hi @jfversluis,
 
Is there any news on this? Recording audio is the main feature of our app. We really need this.

Maui doesn't have recording audio feature yet and there isn't any plug in available. This presents show stopper problem at least for us. We need it Android and IOS platforms.

If it is not going to be available soon, what is the road map?

Thanks

**Comment by bijington** [comment]
@hercul1017 I actually messaged Gerald this morning to say I would try and get this PR completed this week 😉

**Comment by hercul1017** [comment]
That would be awesome.
Thanks,

**Comment by jfversluis** [comment]
@hercul1017 I just want to add to this: I appreciate your enthusiasm on this. But I'd like to point out that this is a community effort by Shaun and myself. We're working on this whenever our personal situation allows for some time.

I understand that this might be a showstopper, however, it shouldn't have to be. If we can write this code and wrap it in a plugin, so can you. A plugin is just someone else doing work that you could've done yourself. This project is open-source, so you can also even contribute back and not only help yourself but also others.

Coincidentally Shaun did mention he's going to look at it this week. So hopefully we'll be able to release it soon!

Thank you for your understanding.

**Comment by hercul1017** [comment]
@jfversluis sorry for the misunderstanding. Somehow I thought this development belongs to Microsoft and it is not a community effort. It is shame that Microsoft didn't think audio feature (playing sound, recording etc.) is not important enough to be part of Maui first release and did not put some resources on it.

I really appreciate competent developers like you and @bijington to do this for community. Thanks again for your time and effort.

**Comment by cdavidyoung** [comment]
I am also confused because I tested this and have been using it for months now.  I thought I did a pull request.  Is there something else I should have done to have successfully gotten this merged back into the main branch?

https://github.com/cdavidyoung/Plugin.Maui.Audio

**Comment by jfversluis** [comment]
@cdavidyoung sorry if that wasn't clear. The functionality is great and your PR is very much appreciated and we will definitely use that. However, as pointed out in said PR, there were a couple of things from a technical/code standpoint that we wanted to fix up and Shaun has proposed to take that to not burden you any further.

Unfortunately he hasn't found to the time to do so yet, but he's committed to make that happen this week.

**Comment by cdavidyoung** [comment]
Thanks, Gerald!  This was a nice introduction to nugets for me.  I have since moved on to another with great success.  This is really an amazing community for sharing code.

https://github.com/cdavidyoung/FFMpegCore

**Comment by hercul1017** [comment]
Is there any news on this project?

**Comment by bijington** [comment]
Sorry it's all on me right now. We have iOS and macOS working under the API surface that we want but Android and Windows still requires effort sadly. 

I will be away for 2 weeks but when I get back this is at the top of my list of things to look at. Although last time I promised Gerald I would look into this my dog sadly passed so I won't be making any promises. I do really hope to wrap it up soon though

**Comment by hercul1017** [comment]
Thanks for quick response @jfversluis  and @bijington . Sorry to hear about your dog.

I believe IOS and Android solution would be good enough for most of the people who need this feature.

Thank you for your efforts.

**Comment by cdavidyoung** [comment]
Just as a reminder, while waiting on the official release, if you don't mind building your own nugets locally you can clone my repo.  This audio recording works on all platforms.

https://github.com/cdavidyoung/Plugin.Maui.Audio

**Comment by hercul1017** [comment]
Hi, is there any news on this project?
